File: test_ai_runner.php
<?php

if (php_sapi_name() === 'cli') {
    $baseUrl = "http://127.0.0.1/Tourist%20Guide%20System/api";
} else {
    $protocol = isset($_SERVER['HTTPS']) && $_SERVER['HTTPS'] === 'on' ? 'https' : 'http';
    $host = $_SERVER['HTTP_HOST'] ?? 'localhost';
    $scriptPath = $_SERVER['SCRIPT_NAME'] ?? '/Tourist Guide System/test_ai_runner.php';
    $baseDir = dirname($scriptPath);
    $baseUrl = "$protocol://$host" . ($baseDir === '/' ? '' : $baseDir) . "/api";
}

$loginUrl = "$baseUrl/login.php";
$chatUrl = "$baseUrl/chat_ai.php";

$username = "test_ai_user";
$password = "password123";

$cookieFile = tempnam(sys_get_temp_dir(), 'cookie');

function makeRequest($url, $data) {
    global $cookieFile;
    
    $ch = curl_init($url);
    curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
    curl_setopt($ch, CURLOPT_POST, true);
    curl_setopt($ch, CURLOPT_POSTFIELDS, json_encode($data));
    curl_setopt($ch, CURLOPT_HTTPHEADER, ['Content-Type: application/json']);
    curl_setopt($ch, CURLOPT_COOKIEJAR, $cookieFile);
    curl_setopt($ch, CURLOPT_COOKIEFILE, $cookieFile);
    
    $response = curl_exec($ch);
    $httpCode = curl_getinfo($ch, CURLINFO_HTTP_CODE);
    $curlError = curl_error($ch);
    curl_close($ch);
    
    if ($httpCode !== 200) {
        echo "Error: HTTP $httpCode - $curlError\n";
        if ($response) echo "Response: " . substr($response, 0, 500) . "\n";
        return null;
    }
    
    $decoded = json_decode($response, true);
    if ($decoded === null && !empty($response)) {
        echo "Error: Invalid JSON response\n";
        echo "Raw Response: " . substr($response, 0, 500) . "\n";
    }
    return $decoded;
}

// Login
echo "Logging in as $username...\n";
$loginData = ['username' => $username, 'password' => $password];
$loginRes = makeRequest($loginUrl, $loginData);

if (!$loginRes || !$loginRes['success']) {
    echo "Login failed: " . ($loginRes['message'] ?? 'Unknown error') . "\n";
    exit(1);
}
echo "Login successful.\n";

$testCases = [
    "Hello",
    "Who are you?",
    "What is this website?",
    "Top rated spots",
    "Where to eat?",
    "Tell me about Laguna",
    "How to get there?",
    "Events in Laguna",
    "Is there a festival?",
    "Where is Rizal Shrine?",
    "Rizal Shrine",
    "tell me about rizal shrine",
    "telll me abouut rizaaal shrin", // Typo test
    "swimming pools",
    "resorts",
    "contact admin",
    "logout",
    "login",
    "how to rate",
    "latest news about laguna",
    "price of buko pie",
    "meaning of life"
];

$results = [];

foreach ($testCases as $msg) {
    echo "\n--- Testing: '$msg' ---\n";
    $res = makeRequest($chatUrl, ['message' => $msg]);
    
    if ($res) {
        $reply = $res['reply'] ?? 'No reply field';
        $confidence = $res['confidence'] ?? 'N/A';
        echo "Reply: " . substr(str_replace("\n", " ", $reply), 0, 100) . "...\n";
        echo "Confidence: $confidence\n";
        
        $results[] = [
            'input' => $msg,
            'reply' => $reply,
            'success' => $res['success'] ?? false
        ];
    } else {
        echo "No response.\n";
        $results[] = [
            'input' => $msg,
            'reply' => 'ERROR',
            'success' => false
        ];
    }
}

echo "\n\n=== Test Summary ===\n";
foreach ($results as $r) {
    $status = $r['success'] ? "PASS" : "FAIL";
    $shortReply = substr(str_replace("\n", " ", $r['reply']), 0, 50);
    echo "[$status] '{$r['input']}' -> $shortReply...\n";
}

// Clean up
unlink($cookieFile);
?>

File: test_ai_runner.py
import requests
import json
import sys

import os

# Base URL logic to be more portable
BASE_URL = os.environ.get("BASE_URL", "http://localhost/Tourist Guide System/api")
LOGIN_URL = f"{BASE_URL}/login.php"
CHAT_URL = f"{BASE_URL}/chat_ai.php"

USERNAME = "test_ai_user"
PASSWORD = "password123"

session = requests.Session()

def login():
    print(f"Logging in as {USERNAME}...")
    try:
        payload = {"username": USERNAME, "password": PASSWORD}
        response = session.post(LOGIN_URL, json=payload)
        response.raise_for_status()
        data = response.json()
        if data.get("success"):
            print("Login successful.")
            return True
        else:
            print(f"Login failed: {data.get('message')}")
            return False
    except Exception as e:
        print(f"Login error: {e}")
        return False

def ask_ai(message):
    try:
        payload = {"message": message}
        response = session.post(CHAT_URL, json=payload)
        response.raise_for_status()
        try:
            data = response.json()
            return data
        except json.JSONDecodeError:
            print(f"Failed to decode JSON: {response.text[:200]}")
            return None
    except Exception as e:
        print(f"Error asking AI: {e}")
        return None

def run_tests():
    if not login():
        return

    test_cases = [
        "Hello",
        "Who are you?",
        "What is this website?",
        "Top rated spots",
        "Where to eat?",
        "Tell me about Laguna",
        "How to get there?",
        "Events in Laguna",
        "Is there a festival?",
        "Where is Rizal Shrine?",
        "Rizal Shrine",
        "tell me about rizal shrine",
        "telll me abouut rizaaal shrin", # Typo test
        "swimming pools",
        "resorts",
        "contact admin",
        "logout",
        "login",
        "how to rate",
        "latest news about laguna", # Should trigger news check/web search logic
        "price of buko pie", # Should trigger web search check
        "meaning of life", # Random fallback
    ]

    results = []

    for msg in test_cases:
        print(f"\n--- Testing: '{msg}' ---")
        response = ask_ai(msg)
        if response:
            reply = response.get('reply', 'No reply field')
            debug = response.get('debug', [])
            confidence = response.get('confidence', 'N/A')
            print(f"Reply: {reply}")
            print(f"Confidence: {confidence}")
            # print(f"Debug: {debug}")
            
            results.append({
                "input": msg,
                "reply": reply,
                "confidence": confidence,
                "success": response.get('success', False)
            })
        else:
            print("No response or error.")
            results.append({
                "input": msg,
                "reply": "ERROR",
                "success": False
            })

    # Summary
    print("\n\n=== Test Summary ===")
    for r in results:
        status = "PASS" if r['success'] else "FAIL"
        print(f"[{status}] '{r['input']}' -> {r['reply'][:50]}...")

if __name__ == "__main__":
    run_tests()
